src/Business/Cognitive/CognitiveMetrics.php

```diff
@@ -89,12 +100,15 @@ private function setRequiredMetricProperties(array $metrics): void
      */
     private function setOptionalMetricProperties(array $metrics): void
     {
-        foreach ($this->metrics as $metricName) {
-            $property = $metricName . 'Weight';
-            if (array_key_exists($property, $metrics)) {
-                $this->$property = $metrics[$property];
-            }
-        }
+        // Not pretty to set each but more efficient than using a loop and $this->metrics
```

Comment:
It's not only not so pretty, it also changes the behavior, doesn't it? Perviously it checked if the prop in metrics array exists and if yes set it, so you could have passed deltas to change only like 2 metrics, with your change you always need to put a full set of $metrics, because every not included property will be set to 0.0. That's probably not an issue in this case because this method probably is only called once?